homeassistant/components/shelly/__init__.py

```diff
@@ -144,16 +182,47 @@ def __init__(self, hass, entry, device: aioshelly.Device):
 
         self.device.subscribe_updates(self.async_set_updated_data)
 
-        self._async_remove_input_events_handler = self.async_add_listener(
-            self._async_input_events_handler
+        self._async_remove_device_updates_handler = self.async_add_listener(
+            self._async_device_updates_handler
         )
         self._last_input_events_count = dict()
 
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self._handle_ha_stop)
 
     @callback
-    def _async_input_events_handler(self):
-        """Handle device input events."""
+    def _async_device_updates_handler(self):
+        """Handle device updates."""
+        if not self.device.initialized:
+            return
+
+        # Restored device initialized, update entities
+        if self.restored_device:
+            _LOGGER.debug("Restoring Shelly Device - %s", self.name)
+            for block in self.device.blocks:
+                for sensor_id in block.sensor_ids:
+                    description = self.hass.data[DOMAIN][DATA_SENSORS]["sensor"].get(
```

Comment:
This seems weird. Why wouldn't we do it the other way around -> when a sensor entity is first initialized, it will calculate this info. That way we don't have to pre-calculate it for everything.

There is now a really weird reliance:

Entity relies on Device Wrapper
Device Wrapper needs entity storage in hass.data

Reply:
Move this all to `async_added_to_hass` lifecycle callback for the entity in case the entity belongs to a sleepy device.

Reply:
Removed, not using anymore, restoring inside entity and not needing the sensors list